src/nav/BackNavigationHandler.js

```diff
@@ -1,33 +0,0 @@
-/* @flow strict-local */
-import type { Node as React$Node } from 'react';
-import { PureComponent } from 'react';
-import { BackHandler } from 'react-native';
-
-import NavigationService from './NavigationService';
-import { navigateBack } from '../actions';
-
-type Props = $ReadOnly<{|
-  children: React$Node,
-|}>;
-
-export default class BackNavigationHandler extends PureComponent<Props> {
-  componentDidMount() {
-    BackHandler.addEventListener('hardwareBackPress', this.handleBackButtonPress);
-  }
-
-  componentWillUnmount() {
-    BackHandler.removeEventListener('hardwareBackPress', this.handleBackButtonPress);
-  }
-
-  handleBackButtonPress = () => {
-    const canGoBack = NavigationService.getState().index > 0;
-    if (canGoBack) {
-      NavigationService.dispatch(navigateBack());
```

Comment:
So this line was actually doing something quite different from this default behavior of React Navigation:

    """
    By default, when user presses the Android hardware back button,
    react-navigation will pop a screen or exit the app if there are no
    screens to pop. This is a sensible default behavior, but there are
    situations when you might want to implement custom handling.
    """

Namely:
```js
export const navigateBack = (): NavigationAction => StackActions.pop({ n: getSameRoutesCount() });
```
It pops all the topmost screens that happen to be on the same route. In particular, if you've navigated from one narrow (route `chat`) to another, it will pop both/all of those narrows.

This is the behavior discussed at #3954, #3503, and other places linked from [here](https://chat.zulip.org/#narrow/stream/48-mobile/topic/priorities/near/995971).

I don't really *like* this behavior, and it might well be an improvement to just drop it in favor of a straightforward "pop one screen" behavior. But we should be sure to consider that explicitly as a UI decision; in this commit it looks like it's happening as an unintended consequence of an infrastructural change.

Reply:
Oh, you're totally right—you had already pointed this out, on a video call, and it seems to have fallen off my radar.